### PR TITLE
Added few Android framework related questions

### PR DESCRIPTION
**Pull request by hardikm9850**
Added few Android framework related questions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,10 +16,24 @@ Android Engineer Interview Questions
 ## Android
  - How does Garbage collection works?
  - What is a dangling pointer?
+ - Explain the terms reference counting, mark and sweep?
```
Comment (anandwana001) [inline README.md:19]: I think we can skip this question as this is something very rare in interviews. The idea of this repository is to collect best possible and most occurring questions.

File: README.md [hunk 2]
```diff
@@ -16,10 +16,24 @@ Android Engineer Interview Questions
...
  - How does Garbage collection works?
  - What is a dangling pointer?
+ - Explain the terms reference counting, mark and sweep? 
  - Elaborate Memory Leak?
  - Explain fragment Lifecycle when it comes to ViewPager and sliding between different fragments.
  - Difference between FragmentStateAdapter and FragmentStatePagerAdapter.
- - 
+ - Difference between Serializable and Parcelable? What are the disadvantages of Serializable?
+ - How you could implement observable SharedPrefs or observable Databases i.e. Observe a certain key/table/query?
+ - How does layout inflation work from xml tags to view reference in memory?
```
Comment (anandwana001) [inline README.md:25]: Ditto

File: README.md [hunk 3]
```diff
@@ -16,10 +16,24 @@ Android Engineer Interview Questions
...
+ - How would you go about implementing findViewById? Given 2 view references, how would you go about finding their ancestor view?
+   https://obvious.in/blog/deflating-the-layoutinflater-working-with-findviewbyid
```
Comment (anandwana001) [inline README.md:27]:
```suggestion
 - How would you go about implementing findViewById? Given 2 view references, how would you go about finding their ancestor view? Article](https://obvious.in/blog/deflating-the-layoutinflater-working-with-findviewbyid)
```

File: README.md [hunk 4]
```diff
@@ -16,10 +16,24 @@ Android Engineer Interview Questions
...
+ - What is a thread, handler, looper and message queue?
```
Comment (anandwana001) [inline README.md:28]:
```suggestion
 - What is a Thread, Handler, Looper and Message Queue?
```

File: README.md [hunk 5]
```diff
@@ -16,10 +16,24 @@ Android Engineer Interview Questions
...
+ - What are the different methods of concurrency on Android? Can you explain the difference between ExecutorService vs CachedThreadPool vs FixedThreadPool vs AsyncTasks vs HandlerThreads?
```
Comment (anandwana001) [inline README.md:29]: We can skip this question

File: README.md [hunk 6]
```diff
@@ -16,10 +16,24 @@ Android Engineer Interview Questions
...
+ - How does ViewModel instance is provided to Activity, Fragment. How does ViewModelProviderStore decide when to retain the instance?
```
Comment (anandwana001) [inline README.md:30]:
```suggestion
 - How does `ViewModel` instance provided to Activity and Fragment. How does `ViewModelProviderStore` decide when to retain the instance?
```

File: README.md [hunk 7]
```diff
@@ -16,10 +16,24 @@ Android Engineer Interview Questions
...
+  Ans : It uses <Key,Value> to hold onto viewmodel instance, these are removed when ViewModelStore#clear is called. 
+  Tip : Look at the accessor of this method to understand how it is retain across configuration changes
```
Comment (anandwana001) [inline README.md:32]: We can skip adding the answer for now. I am planning a different approach to bring the Ans in for all the questions.

File: README.md [hunk 8]
```diff
@@ -16,10 +16,24 @@ Android Engineer Interview Questions
...
+ - How do you inspect and solve Jank issue? [here](https://developer.android.com/studio/profile/jank-detection)
+ -
+   
+
```
Comment (anandwana001) [inline README.md:36]: remove 1 empty line

File: README.md [hunk 9]
```diff
@@ -40,8 +54,13 @@ Android Engineer Interview Questions
...
 ### Networking
-- What is the role of OkHttp and Retroifit?
-- What design pattern is Retroifit?
+- What is the role of OkHttp and Retrofit?
+- What design pattern does Retrofit use?
+- How would optimise handling of access token expiration? How would you handle retry network call when the api fails? (Custom Interceptor response)
+- 
+
+### Webview
+- What are the problems around security when dealing with webviews.
```
Comment (anandwana001) [inline README.md:63]:
```suggestion
- What are the problems around security when dealing with `WebView`.
```

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks for raising the PR. 
Left few thoughts + suggestions.

File: README.md (hunk 1 above)
Comment (hardikm9850) [reply README.md:19]: I was never asked about dangling pointer ever. Perhaps we can remove that.

File: README.md (hunk 2 above)
Comment (hardikm9850) [reply README.md:25]: This is the most common occurring question around data serialisation. I would want to keep it

**Review by anandwana001 — APPROVED** [review]
LGTM. 

Thanks @hardikm9850 for your contribution.

**Pull request merged**